src/main/java/teammates/ui/controller/InstructorCourseRemindAction.java

```diff
@@ -66,7 +66,12 @@ public ActionResult execute() throws EntityDoesNotExistException {
 
             statusToUser.add(new StatusMessage(Const.StatusMessages.COURSE_REMINDER_SENT_TO + studentEmail,
                                                StatusMessageColor.SUCCESS));
-            redirectUrl = Const.ActionURIs.INSTRUCTOR_COURSE_DETAILS_PAGE;
+
+            if (isRequestedFromPage(Const.ActionURIs.INSTRUCTOR_COURSE_DETAILS_PAGE)) {
+                redirectUrl = Const.ActionURIs.INSTRUCTOR_COURSE_DETAILS_PAGE;
+            } else if (isRequestedFromPage(Const.ActionURIs.INSTRUCTOR_STUDENT_LIST_PAGE)) {
+                redirectUrl = Const.ActionURIs.INSTRUCTOR_STUDENT_LIST_PAGE;
+            }
```

Comment:
It would be better to use a request parameter to indicate where to redirect to instead of relying on the referrer.

Reply:
@whipermr5 Thank you, sir. 👍 
I didn't consider this before but it does sound much more stable. I am working on this now.